syntax = "proto3";

package base;

message Empty {
}

message ClientId {
  string id = 1;
}

message SignalId {
  string name = 1;
  NameSpace namespace = 2;
}

message SignalInfo {
  SignalId id = 1;
  MetaData metaData = 2;
}

message Multiplex {
    oneof mode {
        Empty none = 1;
        Empty select = 2;
        int32 filter = 3;
    };
}

message MetaData {
  string description = 4;
  float max = 12;
  float min = 13;
  string unit = 7;
  int32 size = 8;
  bool isRaw = 9;
  double factor = 10;
  double offset = 11;
  repeated string sender = 14;
  repeated string receiver = 15;
  float cycleTime = 16;
  float startValue = 17;
  Multiplex multiplex = 18;
}

message NameSpace {
  string name = 1;
}

message NetworkInfo {
  NameSpace namespace = 1;
  string type = 2;
  string description = 3;
}

message FrameInfo {
  SignalInfo signalInfo = 1;
  repeated SignalInfo childInfo = 2;
}

message Frames {
  repeated FrameInfo frame = 1;
}
